src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -121,19 +153,19 @@ EmberAfNetworkCommissioningError OnAddWiFiNetworkCommandCallbackInternal(app::Co
     {
         if (sNetworks[i].mNetworkType == NetworkType::kUndefined)
         {
-            // Assume they are null terminated, see notes above.
-            size_t ssidLen        = strlen(reinterpret_cast<const char *>(ssid));
-            size_t credentialsLen = strlen(reinterpret_cast<const char *>(credentials));
-
-            VerifyOrExit(ssidLen <= kMaxWiFiSSIDLen, err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_BOUNDS_EXCEEDED);
-            VerifyOrExit(credentialsLen <= kMaxWiFiCredentialsLen, err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_BOUNDS_EXCEEDED);
+            VerifyOrExit(ssidLen <= kMaxWiFiSSIDLen, err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_OUT_OF_RANGE);
+            VerifyOrExit(ssidLen <= kMaxNetworkIDLen, err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_OUT_OF_RANGE);
+            VerifyOrExit(credentialsLen <= kMaxWiFiCredentialsLen, err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_OUT_OF_RANGE);
 
             memcpy(sNetworks[i].mData.mWiFi.mSSID, ssid, ssidLen);
-            sNetworks[i].mData.mWiFi.mSSIDLen = ssidLen;
+            // We have verified that the ssidLen is less than kMaxWiFiSSIDLen.
```

Comment:
Again, can we static_assert that the type of kMaxWiFiSSIDLen is uint8_t?

Reply:
Added some checks above.